DIPs/DIP1005.md

```diff
@@ -0,0 +1,381 @@
+# Dependency-Carrying Declarations
+
+| Field           | Value                                                      |
+|-----------------|------------------------------------------------------------|
+| DIP:            | 1005                                                       |
+| Author:         | Andrei Alexandrescu (andrei@erdani.com)                    |
+| Implementation: | n/a                                                        |
+| Status:         | Draft                                                      |
+
+## Abstract
+
+A Dependency-Carrying Declaration is a D declaration that does not require any `import` declaration to be present outside of it. Such declarations encapsulate their own dependencies, which makes dependency relationships more fine-grained than traditional module- and package-level dependencies.
+
+This DIP makes the argument that Dependency-Carrying Declarations are an essential mechanism for building scalable libraries and applications. It proposes a language addition called "inline import", which allows any function and aggregate D declarations to be transformed into a Dependency-Carrying Declaration.
+
+### Links
+
+[Informal forum review](http://forum.dlang.org/post/o2psvk$1m96$1@digitalmars.com)
+
+### Rationale
```

Comment:
I've had difficulty with this structure. (Putting my organizer hat on: perhaps we could adjust the template?) The usual going in standards proposals and academic circles is like this:

```
## Abstract

Whets the appetite. 

## Introduction

Optional, necessary only for establishing context for involved proposals.

## Motivation/Rationale

Shows how not everything is good with the state of affairs, and how an improvement in that area would help. This should definitely be near the top so as to set up the stage for readers.

## Related Work/Workarounds

This puts things in context and explains how the matter in the Motivation section is being handled by current research and development

## Description 1/2/3

Here work-specific sections are inserted.

## Summary/Conclusion

```

I'm looking at e.g. https://github.com/python/peps/blob/master/pep-0244.txt and http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0306r2.html as related examples to draw from.